### PR TITLE
Show submap ID in rviz

### PR DESCRIPTION
**Pull request by jihoonl**
I can make it toggleable if necessary.

submap_id

### CODE REVIEW

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 1]
```diff
@@ -97,6 +108,13 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
       "" /* title */, visible, "" /* description */, submap_category,
       SLOT(ToggleVisibility()), this);
   submap_node_->attachObject(manual_object_);
+
+  // TODO: Make it toggleable.
+  submap_id_text_.setTextAlignment(::rviz::MovableText::H_CENTER,
+                                   ::rviz::MovableText::V_ABOVE);
+  submap_id_text_node_->setPosition(ToOgre(kSubmapIdHeight));
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: Maybe I misunderstand this, but can't you attach the `submap_id_text_` to the `scene_node_` (could you rebase? this is now called `submap_node_`) and use `submap_id_text_.setLocalTranslation()`?

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 2]
```diff
@@ -97,6 +108,13 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
       "" /* title */, visible, "" /* description */, submap_category,
       SLOT(ToggleVisibility()), this);
   submap_node_->attachObject(manual_object_);
+
+  // TODO: Make it toggleable.
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]:
```
TODO(jihoonl): Make it toggleable.
```
?

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 3]
```diff
@@ -75,10 +80,16 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
                       ->getRootSceneNode()
                       ->createChildSceneNode()),
       submap_node_(scene_node_->createChildSceneNode()),
+      submap_id_text_node_(scene_node_->createChildSceneNode()),
       manual_object_(display_context->getSceneManager()->createManualObject(
           kManualObjectPrefix + GetSubmapIdentifier(id))),
       pose_axes_(display_context->getSceneManager(), scene_node_,
                  pose_axes_length, pose_axes_radius),
+      submap_id_text_(QString("(%1,%2)")
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: Would it make sense to only set the string here and all other properties below? Right now the transform is split between the size here and the translation and alignment below.

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc (hunk 1 above)
Comment (jihoonl) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: I tried both `setLocalTranslation()` and `setGlobalTranslation()`. And they don't seem to show texts on expected position. Having child  node seems like the cleanest way of showing texts on axes above.

**Comment by jihoonl** [comment]
Sycned with upstream and addressed comments.

**Comment by wohe** [comment]
@jihoonl It does not look like the rebase worked, there is a huge diff now.

**Comment by jihoonl** [comment]
should be fixed. now.

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 4]
```diff
@@ -65,6 +75,14 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
   visibility_ = ::cartographer::common::make_unique<::rviz::BoolProperty>(
       "" /* title */, visible, "" /* description */, submap_category,
       SLOT(ToggleVisibility()), this);
+  submap_id_text_.setFontName(kSubmapIdFont);
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: Default? Remove.

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 5]
```diff
@@ -65,6 +75,14 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
...
+  submap_id_text_.setCharacterHeight(kSubmapIdCharHeight);
+  submap_id_text_.setColor(kSubmapIdColor);
+  submap_id_text_.setTextAlignment(::rviz::MovableText::H_CENTER,
+                                   ::rviz::MovableText::V_ABOVE);
+  // TODO(jihoonl) Make it toggleable.
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: Nit: Colon after `TODO(jihoonl)`.

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 6]
```diff
@@ -54,9 +59,14 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
       scene_node_(display_context->getSceneManager()
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: This still does not look like the latest version. There is no `scene_node_` anymore at master. It is named `submap_node_` instead.

File: cartographer_rviz/cartographer_rviz/drawable_submap.cc [hunk 7]
```diff
@@ -53,9 +57,14 @@ DrawableSubmap::DrawableSubmap(const ::cartographer::mapping::SubmapId& id,
     : id_(id),
       display_context_(display_context),
       submap_node_(map_node->createChildSceneNode()),
+      submap_id_text_node_(submap_node_->createChildSceneNode()),
```
Comment (wohe) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: Destroy in the destructor similar to `submap_node_`?

Comment (jihoonl) [inline cartographer_rviz/cartographer_rviz/drawable_submap.cc]: done.

**Pull request merged**